Kconfig.zephyr

```diff
@@ -429,7 +429,9 @@ endchoice
 
 config LTO
 	bool "Link Time Optimization [EXPERIMENTAL]"
-	depends on (!(GEN_ISR_TABLES || GEN_IRQ_VECTOR_TABLE) || ISR_TABLES_LOCAL_DECLARATION) && !NATIVE_LIBRARY
+	depends on (!(GEN_ISR_TABLES || GEN_IRQ_VECTOR_TABLE) || ISR_TABLES_LOCAL_DECLARATION) \
+	  && !NATIVE_LIBRARY \
+	  && !CODE_DATA_RELOCATION
```

Comment:
You can split the outer "&&"'s into separate "depends on" lines for readability.   Also "depends on !X" relationships are IMHO better expressed as a BUILD_ASSERT() in the C code somewhere, because there you document it as a straightforward error ("/* LTO not compatible with NATIVE_LIBRARY */") and not an inscrutable kconfig dependency failure.

Reply:
Will improve, then re-request a review from you.

Reply:
I incorporated the suggestion regarding the multiple "depends on" lines.

Did not do the BUILD_ASSERTS() suggestion, because I'd rather have Kconfig complaining about/enforcing things early on, not delaying the error until the build is happening.